pipeline/snapcraft/snapcraft.go

```diff
@@ -0,0 +1,142 @@
+// Package snapcraft implements the Pipe interface providing Snapcraft bindings.
+package snapcraft
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/goreleaser/goreleaser/context"
+	"golang.org/x/sync/errgroup"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// ErrNoSnapcraft is shown when snapcraft cannot be found in $PATH
+var ErrNoSnapcraft = errors.New("snapcraft not present in $PATH")
+
+// SnapcraftMetadata to generate the snap package
+type SnapcraftMetadata struct {
+	Name          string
+	Version       string
+	Summary       string
+	Description   string
+	Grade         string `yaml:",omitempty"`
+	Confinement   string `yaml:",omitempty"`
+	Architectures []string
```

Comment:
It is used here: https://github.com/goreleaser/goreleaser/pull/317/files#diff-0c1a079410d0d8f77011355f28769eb6R112